auth_keycloak/models/res_users.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from odoo import api, models, exceptions, _
+import logging
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class OAuthError(Exception):
+    pass
+
+# NOTE: `sub` is THE user id in keycloak
+# https://www.keycloak.org/docs/latest/server_development/index.html#_action_token_anatomy  # noqa
+# and you cannot change it https://stackoverflow.com/questions/46529363
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    def _keycloak_validate(self, provider, access_token):
+        """Validate token against Keycloak."""
+        logger.debug('Calling: %s' % provider.validation_endpoint)
+        resp = requests.post(
+            provider.validation_endpoint,
+            data={'token': access_token},
+            auth=(provider.client_id, provider.client_secret)
+        )
+        if not resp.ok:
+            raise OAuthError(resp.reason)
+        validation = resp.json()
+        if validation.get("error"):
+            raise OAuthError(validation)
+        logger.debug('Validation: %s' % str(validation))
+        return validation
+
+    @api.model
+    def _auth_oauth_validate(self, provider, access_token):
+        """Override to use authentication headers.
+
+        The method `_auth_oauth_rpc` is not pluggable
+        as you don't have the provider there.
+        """
+        oauth_provider = self.env['auth.oauth.provider'].browse(provider)
```

Comment:
If you have to browse, means you get an id, so using `provider_id` would make more sense here. Although passing the recordset might avoid the browse...

Reply:
I agree w/ you 100%. Fact is: I'm respecting odoo's [signature](https://github.com/odoo/odoo/blob/10.0/addons/auth_oauth/models/res_users.py#L39) :sob: